app/screens/send-bitcoin-screen/send-bitcoin-destination-screen.tsx

```diff
@@ -102,36 +130,45 @@ const SendBitcoinDestinationScreen: React.FC<Props> = ({ route }) => {
     skip: !isAuthed,
   })
 
-  const wallets = data?.me?.defaultAccount.wallets
-  const bitcoinNetwork = data?.globals?.network
-  const contacts = useMemo(() => data?.me?.contacts ?? [], [data?.me?.contacts])
+  const wallets: Wallets = useMemo(
+    () => data?.me?.defaultAccount.wallets,
+    [data?.me?.defaultAccount.wallets],
+  )
+  const bitcoinNetwork: Network | undefined = useMemo(
+    () => data?.globals?.network,
+    [data?.globals?.network],
+  )
+  const contacts: Contacts = useMemo(() => data?.me?.contacts ?? [], [data?.me?.contacts])
 
   const { LL } = useI18nContext()
   const [accountDefaultWalletQuery] = useAccountDefaultWalletLazyQuery({
     fetchPolicy: "no-cache",
   })
 
-  const validateDestination = useMemo(() => {
+  const willInitiateValidation = React.useCallback(() => {
     if (!bitcoinNetwork || !wallets || !contacts) {
-      return null
+      return false
     }
 
-    return async (rawInput: string) => {
-      if (destinationState.destinationState !== "entering") {
+    dispatchDestinationStateAction({
+      type: SendBitcoinActions.SetValidating,
+      payload: {},
+    })
+    return true
+  }, [bitcoinNetwork, wallets, contacts])
+
+  const validateDestination = React.useCallback(
+    async (rawInput: string, obj: NetworkWalletsContacts) => {
+      // extra check to ensure nothing has changed since initiating validation
```

Comment:
maybe I'm missing something, but in the line below, we are just checking for presence of value, not that the value has changed?

Reply:
You're correct it's just for the presence of value. I pass them as such because of typescript annoyances. In the validateDestination function the values are used a few times and they must be verified to not be undefined (since the type Network that comes from GraphQL for example has the type 'Network | undefined').

Reply:
I've removed them in most instances, but I left them there with a comment why there's two checks. 1 before changing state 'validating', and 1 before actually doing validation.